pkg/rtc/mediatracksubscriptions.go

```diff
@@ -251,21 +252,31 @@ func (t *MediaTrackSubscriptions) AddSubscriber(sub types.Participant, codec web
 // stop all forwarders to the client
 func (t *MediaTrackSubscriptions) RemoveSubscriber(participantID livekit.ParticipantID) {
 	subTrack := t.getSubscribedTrack(participantID)
+
+	t.subscribedTracksMu.Lock()
+	delete(t.subscribedTracks, participantID)
+	t.subscribedTracksMu.Unlock()
```

Comment:
Doing this here while a subscriber is removed because we start a goroutine to close down track and were removing this subscription in the down track close callback.

But, when adding subscriber we were also checking if it is already subscribed. In the edge case of unsubscription followed by a subscription very close to each other, we may not have gotten the down track close callback and we will not process the subscription as already subscribed check will ignore the second subscription.